components/claim/components/ldap_to_email.tsx

```diff
@@ -0,0 +1,233 @@
+                setLdapPasswordError('');
+                setServerError('');
+                return;
+            }
+        }
+
+        const confirmPassword = passwordConfirmInput.current?.value;
+        if (!confirmPassword || password !== confirmPassword) {
+            setConfirmError(localizeMessage('claim.ldap_to_email.pwdNotMatch', 'Passwords do not match.'));
+            setPasswordError('');
+            setLdapPasswordError('');
+            setServerError('');
+            return;
+        }
+
+        setPassword(password);
+        setLdapPassword(ldapPassword);
+
+        if (props.email) {
+            submit({loginId: props.email, password, ldapPasswordParam: ldapPassword});
+        }
+    };
+
+    const submit = ({loginId, password, token = '', ldapPasswordParam}: SubmitOptions) => {
+        props.switchLdapToEmail(ldapPasswordParam || ldapPassword, loginId, password, token).then(({data, error: err}) => {
+            if (data?.follow_link) {
+                window.location.href = data.follow_link;
+            } else if (err) {
+                if (err.server_error_id.startsWith('model.user.is_valid.pwd')) {
+                    setPasswordError(err.message);
+                    setShowMfa(false);
+                } else if (err.server_error_id === ClaimErrors.ENT_LDAP_LOGIN_INVALID_PASSWORD) {
+                    setLdapPasswordError(err.message);
+                    setShowMfa(false);
+                } else if (!showMfa && err.server_error_id === ClaimErrors.MFA_VALIDATE_TOKEN_AUTHENTICATE) {
+                    setShowMfa(true);
+                } else {
+                    setServerError(err.message);
+                    setShowMfa(false);
+                }
+            }
+        });
+    };
+
+    let serverErrorElement: JSX.Element | null = null;
+    let formClass = 'form-group';
+    if (serverError) {
+        serverErrorElement = <div className='form-group has-error'><label className='control-label'>{serverError}</label></div>;
+        formClass += ' has-error';
+    }
+
+    let passwordErrorElement: JSX.Element | null = null;
+    let passwordClass = 'form-group';
+    if (passwordError) {
+        passwordErrorElement = <div className='form-group has-error'><label className='control-label'>{passwordError}</label></div>;
+        passwordClass += ' has-error';
+    }
+
+    let ldapPasswordErrorElement: JSX.Element | null = null;
+    let ldapPasswordClass = 'form-group';
+    if (ldapPasswordError) {
+        ldapPasswordErrorElement = <div className='form-group has-error'><label className='control-label'>{ldapPasswordError}</label></div>;
+        ldapPasswordClass += ' has-error';
+    }
+
+    let confirmErrorElement: JSX.Element | null = null;
+    let confimClass = 'form-group';
+    if (confirmError) {
+        confirmErrorElement = <div className='form-group has-error'><label className='control-label'>{confirmError}</label></div>;
+        confimClass += ' has-error';
+    }
```

Comment:
same comment as in the `email_to_ldap.tsx` file

Reply:
Done, thanks